packages/jest-types/__typetests__/expect.test.ts

```diff
@@ -180,6 +180,21 @@ expectType<void>(expect({a: 1, b: 2}).toMatchObject({b: 2}));
 expectType<void>(
   expect([{a: 1}, {b: 2, c: true}]).toMatchObject([{a: 1}, {b: 2}]),
 );
+expectType<void>(
+  expect([{a: 1}, {b: 2, c: true}]).toMatchObject([{a: 1}, {b: 2}], {
+    ignoreElementsOrder: true,
+  }),
+);
```

Comment:
Type tests will be only compiled, but never executed. So this case is exactly the same as below.

```suggestion
expectType<void>(
  expect([{a: 1}, {b: 2, c: true}]).toMatchObject([{a: 1}, {b: 2}], {
    ignoreElementsOrder: true,
  }),
);
```